analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -0,0 +1,126 @@
+        }
+
+        [TestMethod]
+        public void SimpleAssignment_ToLocalVariable_FromTrackedSymbol_Chained()
+        {
+            var setter = new LiteralDummyTestCheck();
+            var collector = SETestContext.CreateCS(@"var a = true; bool b,c; c = b = a; Tag(""c"", c);", setter).Collector;
+            collector.Validate("Literal: true", x => x.State[x.Operation].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.Validate("SimpleAssignment: a = true (Implicit)", x => x.State[((ISimpleAssignmentOperation)x.Operation.Instance).Target].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.ValidateTag("c", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void SimpleAssignment_ToParameter_FromLiteral()
+        {
+            var setter = new LiteralDummyTestCheck();
+            var collector = SETestContext.CreateCS(@"boolParameter = true; Tag(""boolParameter"", boolParameter);", setter).Collector;
+            collector.Validate("Literal: true", x => x.State[x.Operation].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.Validate("SimpleAssignment: boolParameter = true", x => x.State[((ISimpleAssignmentOperation)x.Operation.Instance).Target].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.ValidateTag("boolParameter", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void SimpleAssignment_ToLocalVariable_FromTrackedSymbol_CS()
+        {
+            var setter = new PreProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.Kind == OperationKind.ParameterReference)
+                {
+                    x.State[x.Operation].SetConstraint(DummyConstraint.Dummy);
+                    return x.State.SetSymbolValue(x.Operation.Instance.TrackedSymbol(), x.State[x.Operation]);
+                }
+                return x.State;
+            });
+            var collector = SETestContext.CreateCS(@"var b = boolParameter; Tag(""b"", b);", setter).Collector;
+            collector.ValidateTag("b", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void SimpleAssignment_ToLocalVariable_FromTrackedSymbol_VB()
+        {
+            var setter = new PreProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.Kind == OperationKind.ParameterReference)
+                {
+                    x.State[x.Operation].SetConstraint(DummyConstraint.Dummy);
+                    return x.State.SetSymbolValue(x.Operation.Instance.TrackedSymbol(), x.State[x.Operation]);
+                }
+                return x.State;
+            });
+            var collector = SETestContext.CreateVB(
+@"BoolParameter = True
+Dim B As Boolean = BoolParameter
```

Comment:
This is not symmetric with the C# version
```suggestion
@"Dim B As Boolean = BoolParameter
```